File: 8_Kyu_Kata/find-multiples-of-a-number/Python/find_multiples_of_a_number_test.py
# [object Object] - Find Multiples of a Number  [ ID: 58ca658cc0d6401f2700045f  (find-multiples-of-a-number) ]
# URL: https://www.codewars.com/kata/58ca658cc0d6401f2700045f
# Category: REFERENCE  |  Tags: FUNDAMENTALS
# ******************************************************************************
import codewars_test as test
from solution import find_multiples

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(find_multiples(5, 25), [5, 10, 15, 20, 25])
        test.assert_equals(find_multiples(1, 2), [1, 2])
        test.assert_equals(find_multiples(5, 7), [5])
        test.assert_equals(find_multiples(4, 27), [4, 8, 12, 16, 20, 24])
        test.assert_equals(find_multiples(11, 54), [11, 22, 33, 44])
    
@test.describe("Random Tests:")
def random_tests():    
    import random
    
    def my_find_multiples(integer, limit):
        return [i for i in range(integer, limit + 1) if i % integer == 0]

    for i in range(50):
        integer = random.randint(1,100)
        limit = random.randint(101,30000)
        solution = my_find_multiples(integer, limit)
        @test.it(f"testing for find_multiples({integer}, {limit})")
        def test_case():
             test.assert_equals(find_multiples(integer, limit), solution)
    

File: 8-kyu-kata/find-multiples-of-a-number/javascript/findMultiplesOfANumber.Test.js
// 8 kyu - Find Multiples of a Number  [ ID: 58ca658cc0d6401f2700045f  (find-multiples-of-a-number) ]
// URL: https://www.codewars.com/kata/58ca658cc0d6401f2700045f
// Category: REFERENCE  |  Tags: FUNDAMENTALS
// *****************************************************************************
describe("Basic Tests", () => {
  Test.assertSimilar(findMultiples(5, 25), [5, 10, 15, 20, 25])
  Test.assertSimilar(findMultiples(1, 2), [1, 2])
  Test.assertSimilar(findMultiples(5, 7), [5])
  Test.assertSimilar(findMultiples(4, 27), [4, 8, 12, 16, 20, 24])
  Test.assertSimilar(findMultiples(11, 54), [11, 22, 33, 44])
})

describe("Randomt Tests", () => {
  const randint = (a, b) => ~~(Math.random() * (b - a + 1) + a)
  const sol = (n, l) => Array.from({ length: ~~(l / n) }, (a, i) => i * n + n)

  for (let i = 0; i < 40; i++) {
    const n = randint(1, 20),
      l = n + randint(1, Math.pow(10, randint(0, 4)))
    it(`Testing for ${n} and ${l}`, () => {
      Test.assertSimilar(findMultiples(n, l), sol(n, l), "It should work for random inputs too")
    })
  }
})
